var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -464,22 +499,26 @@ def cmake_args(self):
             else:
                 options.extend(
                     [
-                        "-DCMAKE_C_COMPILER=%s" % spack_cc,
-                        "-DCMAKE_CXX_COMPILER=%s" % spack_cxx,
+                        "-DCMAKE_C_COMPILER=%s" % gmx_cc,
+                        "-DCMAKE_CXX_COMPILER=%s" % gmx_cxx,
                         "-DMPI_C_COMPILER=%s" % self.spec["mpi"].mpicc,
                         "-DMPI_CXX_COMPILER=%s" % self.spec["mpi"].mpicxx,
                     ]
                 )
         else:
             options.extend(
                 [
-                    "-DCMAKE_C_COMPILER=%s" % spack_cc,
-                    "-DCMAKE_CXX_COMPILER=%s" % spack_cxx,
+                    "-DCMAKE_C_COMPILER=%s" % gmx_cc,
+                    "-DCMAKE_CXX_COMPILER=%s" % gmx_cxx,
                     "-DGMX_MPI:BOOL=OFF",
                     "-DGMX_THREAD_MPI:BOOL=ON",
                 ]
             )
 
+        # Here we cannot use spack_cc because we need also libstdc++ to be reachable
+        # Spack wrapper (spack_cc) hides includes/lib and CMake will fail
+        options.append("-DGMX_GPLUSPLUS_PATH=%s" % self.pkg.compiler.cxx)
```

Comment:
(compare) #41767 sets `-DHIP_CXX_COMPILER=` (in `hipsycl`) to handle this same problem, although I didn't have to do it in `gromacs`, perhaps that's an issue with building `@2024`